app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -219,8 +219,9 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             it.start()
         }
 
-        if (isActivityColdStarted(intent, savedInstanceState) &&
-                !externalSourceIntentProcessors.any { it.process(intent, navHost.navController, this.intent) }) {
+        if (!settings().shouldStartOnHome() &&
+            isActivityColdStarted(intent, savedInstanceState) &&
+            !externalSourceIntentProcessors.any { it.process(intent, navHost.navController, this.intent) }) {
```

Comment:
I wish there was a cleaner way to write this condition 🤔

Reply:
What do you think about creating a helper function call `shouldNavigateToBrowserFragmentOnCouldStart()`?

Reply:
It can either be a variable (if you don't think it's gonna be call often in other places, and prefer to keep it within the same code block where it is used) or function call. I think it's probably worthwhile to add a comment as well.